devtools/running_packages.rst

```diff
@@ -114,6 +113,227 @@ with:
+--------------------
+
+The more advanced approach is to use the :ref:`json generator <json_generator>`. First off, invoke the conan to generate the ``conanbuildinfo.json`` 
+file containing all required information about dependencies:
+
+.. code-block:: bash
+
+    $ conan install . -g json
+
+The file produced is fully machine-readable and could be used by scripts to prepare the distribution.
+
+.. code-block:: python
+
+        import os
+        import json
+
+        data = json.load(open("conanbuildinfo.json"))
+
+        dep_lib_dirs = dict()
+        dep_bin_dirs = dict()
+
+        for dep in data["dependencies"]:
+            root = dep["rootpath"]
+            lib_paths = dep["lib_paths"]
+            bin_paths = dep["bin_paths"]
+
+            dep_lib_dirs = dict()
+            dep_bin_dirs = dict()
+
+            for lib_path in lib_paths:
+                if os.listdir(lib_path):
+                    lib_dir = os.path.relpath(lib_path, root)
+                    dep_lib_dirs[lib_path] = lib_dir
+            for bin_path in bin_paths:
+                if os.listdir(bin_path):
+                    bin_dir = os.path.relpath(bin_path, root)
+                    dep_bin_dirs[bin_path] = bin_dir
+
+Then, the information collected might be used to copy the required files into the destination:
+
+.. code-block:: python
+
+    from distutils.dir_util import copy_tree
+
+    if not os.path.isdir(destination):
+        os.makedirs(destination)
+    for src_lib_dir, dst_bin_dir in dep_lib_dirs.items():
+        copy_tree(src_lib_dir, os.path.join(destination, dst_lib_dir))
+    for src_bin_dir, dst_bin_dir in dep_bin_dirs.items():
+        copy_tree(src_bin_dir, os.path.join(destination, dst_bin_dir))
+
+The advantage over the ``deploy`` generator is fine-grained control: here we copy only binaries and libraries.
+It's also could be easily modified to apply some sort of filtering (e.g. to copy only shared libraries, 
+and omit any static libraries or auxiliary files such as pkg-config .pc files).
+The extracted information may also be used to generate a simple startup script (as described below):
+
+.. code-block:: python
+
+    executable = "MyApp"  # just an example
+    varname = "$APPDIR"
+
+    def _format_dirs(dirs):
+        return ":".join(["%s/%s" % (varname, d) for d in dirs])
+
+    path = _format_dirs(bin_dirs.values())
+    ld_library_path = _format_dirs(bin_dirs.values())
+    exe = varname + "/" + executable
+
+    content = """#!/usr/bin/env bash
+    set -ex
+    export PATH=$PATH:{path}
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:{ld_library_path}
+    pushd $(dirname {exe})
+    $(basename {exe})
+    popd
+    """.format(path=path,
+           ld_library_path=ld_library_path,
+           exe=exe)
+
+.. note::
+
+    The full example might be found on `GitHub <https://github.com/SSE4/conan-deploy-tool>`_.
+
+.. _deployment_challenges:
+
+Deployment challenges
+---------------------
+
+C standard library
```

Comment:
Have you tried to build the pages? you are putting everything at the same section level, right?

Reply:
changed the formatting to use `~~~~~~~~~~` as 3rd level heading